epidose/back_end/cuckoo_filter_digital_signature.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/sh
+#
+# Script to generate private/public keys and sign Cuckoo filters
+#
+# Copyright 2020 Stefanos Georgiou
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+export APP_NAME=sign_filter
+
+# Path towards the private key
+PRIVATE_KEY="/var/lib/epidose/privateFilter.pem"
+
+# Path towards the public key
+PUBLIC_KEY="/var/lib/epidose/publicFilter.pem"
+
+# Path of the signature
+SIGNATURE_PATH="/var/lib/epidose/signature.sha256"
+
+# Log with a timestamp
+log()
+{
+  # Output is redirected to the log file if needed at the script's lop level
+  date +'%F %T ' | tr -d \\n 1>&2
+  echo "$@" 1>&2
+}
+
+# Report usage information and exit with an error
+usage()
+{
+cat <<EOF 1>&2
+Usage: $0 [-s] fileToSign [-p] passphrase
+-s, --sign "filterPath"		Sign theCuckoo filter
+-p, --passphrase "secret" 	Passphrase for the private key
+-v, --verify "filterPath"	Verify signed filter
+-h, --help			Usage instructions
+EOF
+exit 1
+}
+
+# Check if executed as sudo user
```

Comment:
You mean `root` user.  We don't check this, we assume that we run with the appropriate permissions and use `set -e` to fail if the assumption doesn't hold.